modules/37_industry/subsectors/equations.gms

```diff
@@ -242,6 +242,21 @@ q37_feedstocksLimit(ttot,regi,entySE,entyFE,emiMkt)$(
   vm_demFENonEnergySector(ttot,regi,entySE,entyFE,"indst",emiMkt)
 ;
 
+*' Feedstocks have identical fossil/biomass/synfuel shares as industry FE
+q37_feedstocksShares(t,regi,entySE,entyFE,emiMkt)$(
+                                          sum(te, se2fe(entySE,entyFE,te)) ) ..
+    vm_demFEsector_afterTax(t,regi,entySE,entyFE,"indst",emiMkt)
+  * sum(se2fe(entySE2,entyFE,te),
+      vm_demFENonEnergySector(t,regi,entySE2,entyFE,"indst",emiMkt)
+    )
+  =e=
+    vm_demFENonEnergySector(t,regi,entySE,entyFE,"indst",emiMkt)
+  * sum(se2fe2(entySE2,entyFE,te),
+      vm_demFEsector_afterTax(t,regi,entySE2,entyFE,"indst",emiMkt)
+    )
+;
```

Comment:
@Renato-Rodrigues what is your take on this formulation, w.r.t. CONOPT solvability?

Reply:
> @Renato-Rodrigues what is your take on this formulation, w.r.t. CONOPT solvability?

I would say it is not ideal, as it enforces strong circularities in two levels: 
(1) the circularity between an specific FE carrier in industry (`vm_demFEsector(entyFE)`) and a sub-product of it (`vm_demFENonEnergySector(entyFE)`), as the endogenous share calculated at `q37_feedstocksShares` will influence the value of the first variable marginals.
(2) the dependency of an specific FE carrier in industry (`vm_demFEsector(entyFE)`) and the total FE industry. Changing any FE carrier marginal, changes all other FE carrier marginal values.

This is taken into account in the Jacobian matrix, but it demands at least two additional second order terms that could be quite complicated depending of the case.

I am not sure although much complex this is to the model to solve without testing it.

Ideally, feedstocks demand should be treated independently of the total industry final energy demand to avoid this. I imagine this could remove the need for this circularity, but this would require reworking more equations, so if the solution time is acceptable with this solution, I guess we would have to live with that for now.